src/clr-addons/treetable/treetable-row.ts

```diff
@@ -4,43 +4,51 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { AfterContentInit, Component, ContentChildren, Input, QueryList } from '@angular/core';
+import { Component, Input } from '@angular/core';
+import { animate, style, transition, trigger } from '@angular/animations';
 
 @Component({
   selector: 'clr-tt-row',
   templateUrl: './treetable-row.html',
   styleUrls: ['./treetable-row.scss'],
+  animations: [
+    trigger('collapseExpandAnimation', [
+      transition(':enter', [
+        style({ opacity: 0, height: 0, overflow: 'hidden' }),
+        animate('300ms', style({ opacity: 1, height: '*' })),
+      ]),
+      transition(':leave', [
+        style({ opacity: 1, height: '*', overflow: 'hidden' }),
+        animate('300ms', style({ opacity: 0, height: 0 })),
+      ]),
+    ]),
+  ],
 })
-export class ClrTreetableRow implements AfterContentInit {
-  @Input() clrExpanded = false;
-  @Input() clrClickable = true;
+export class ClrTreetableRow {
+  @Input('clrExpanded') expanded = false;
+  @Input('clrClickable') clickable = true;
 
-  isExpandable = false;
-
-  @ContentChildren(ClrTreetableRow) ttRows: QueryList<ClrTreetableRow>;
-
-  ngAfterContentInit(): void {
-    this.initIsExpandable();
-    this.ttRows.changes.subscribe(() => this.initIsExpandable());
-  }
-
-  private initIsExpandable(): void {
-    this.isExpandable = this.ttRows.length > 1;
-  }
+  @Input('clrExpandable') expandable = false;
```

Comment:
do we have a usecase where the row has children, but should not be expandable?
Because this way you have to specify expandable on every row.

If we have such a use case, I bet this is the rarer case, so the default value should be true and should only be set to false for those exceptional cases.